test/performance/odp_dma_perf.c

```diff
@@ -854,7 +854,7 @@ static void run_transfers_mt_safe(sd_t *sd, stats_t *stats)
 	for (uint32_t i = 0U; i < count; ++i) {
 		info = &infos[i];
 
-		if (odp_ticketlock_trylock(&info->lock)) {
+		if (odp_ticketlock_trylock(&info->lock) == 1) {
```

Comment:
I think this patch could be dropped. The code already works according to the ODP spec which says that the trylock function returns 0 or 1. So this change does not fix anything, contrary to what the commit title says. Only in the case the trylock returns something outside the spec this might change the behaviour by I do not see why this would be more correct than e.g. checking that the return value is not zero. Asserting that the return value is either 0 or 1 would make sense in a validation test, but a perf test can pretty much assume that ODP functions do what they say. So I think the existing code is best here.

Reply:
I would agree if OK case was specified as ">0" or return type was bool, however the type is signed integer and OK case is exactly 1. IMO, also "application" code from ODP repo should (try) to display exemplary usage of the API.

Reply:
Democratically dropped in v4.